index.bs

```diff
@@ -2166,27 +2175,172 @@ interface MLGraph {
+                    1. Set the dimensions of |inputTensor| to |value|.{{MLArrayInput/dimensions}}.
+                1. Else:
+                    1. Set the dimensions of |inputTensor| to |inputDesc|.{{MLOperandDescriptor/dimensions}}.
+                1. If |value| is an {{MLArrayInput}}, then:
+                    1. Set the values of |inputTensor| to the values of |value|.{{MLArrayInput/resource}}.
+                1. If |value| is an {{ArrayBufferView}}, then:
+                    1. Set the values of |inputTensor| to the values of |value|.
+                1. Set the input of |graph|.{{MLGraph/[[implementation]]}} that is associated with |key| to |inputTensor|.
+            1. For each |key| -> |value| of |outputs|:
+                1. Issue a compute request for output of |graph|.{{MLGraph/[[implementation]]}} that is associated with |key|.
+                1. Wait for the compute request to be completed.
+                1. If there is an error returned by |graph|.{{MLGraph/[[implementation]]}}, then:
+                    1. Throw an {{OperationError}} {{DOMException}} and stop.
+                1. Else:
+                    1. Let |outputTensor| be the output tensor returned by |graph|.{{MLGraph/[[implementation]]}}.
+                    1. If the kind of |value| is not compatible with the value type of |outputTensor|, then throw a {{DataError}} {{DOMException}} and stop.
+                    1. Let |outputSize| be 1.
+                    1. For each |dimension| of dimensions of |outputTensor|:
+                        1. Set |outputSize| to the product of |outputSize| and |dimension|.
+                    1. If |outputSize| is greater than the length of |value|, then:
+                        1. Throw a {{DataError}} {{DOMException}} and stop.
+                    1. Else:
+                        1. Set the values of |value| to the values of |outputTensor|.
+            1. Return {{undefined}}.
+        </div>
+</dl>
+
+## MLAwaitedExecution ## {#api-mlawaitedexecution}
+The {{MLAwaitedExecution}} interface represents a method of execution that asynchronously carries out the computational workload of a compiled graph {{MLGraph}} on a worker thread to avoid blocking the calling thread while producing results as defined by the operations in the graph. This method of execution requires an {{MLContext}} created with {{MLContextOptions}} or {{WebGLRenderingContext}}.
+
+<script type=idl>
+typedef (MLBufferView or WebGLTexture or GPUTexture) MLResource;
+
+dictionary MLInput {
+  required MLResource resource;
+  required sequence<long> dimensions;
+};
+
+typedef record<DOMString, (MLResource or MLInput)> MLNamedInputs;
+typedef record<DOMString, MLResource> MLNamedOutputs;
+
+[SecureContext, Exposed=(Window, DedicatedWorker)]
+interface MLAwaitedExecution {
+  constructor(MLContext context);
+
+  Promise<MLNamedOutputs> compute(MLGraph graph, MLNamedInputs inputs, MLNamedOutputs outputs);
```

Comment:
My understanding is that `Promise<Return-Type>` is a semantic used for a deferred async result on the return value. In this case, the result will be placed on the `outputs` param, which should also be the return value on the return type. Let me know if I get it wrong.

Reply:
`Promise<Return-Type>` means that a developer would write code à la 
```js
const outputs = await executor.compute(graph, inputs, namedoutputs);
```
I agree with @huningxin that this feels wrong, since the expectation is instead that namedoutputs will be modified in place by the compute method - having the same object also returns via the promise is at best confusing.